classes/Tools.php

```diff
@@ -4307,6 +4309,40 @@ public static function isFileFresh(string $file, int $timeout = self::CACHE_LIFE
 
         return false;
     }
+
+    /**
+     * @return bool
+     */
+    public static function isDetectCountryFromBrowserLanguageAvailable(): bool
+    {
+        $languagesAvailable = self::languagesAvailable();
+
+        if (!isset($languagesAvailable[0])) {
+            return false;
+        }
+
+        return Configuration::get('PS_DETECT_COUNTRY') && isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])
+            && self::isValidAcceptLanguageAvailable()
+            && Validate::isLanguageIsoCode($languagesAvailable[0]);
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isValidAcceptLanguageAvailable(): bool
+    {
+        return preg_match(self::LANGUAGE_EXTRACTOR_REGEXP, $_SERVER['HTTP_ACCEPT_LANGUAGE'], $languages);
+    }
+
+    /**
+     * @return array
+     */
+    public static function languagesAvailable(): array
```

Comment:
```suggestion
    public static function getLanguageFromHttpHeader(): array
```

Reply:
are you agree with getLanguagesFromHttpHeader because it returns an array, I fact I think that it never returns more than one item.... so what about :
public static function getLanguageFromHttpHeader(): ?string

Reply:
Since we are searching for a Country iso code I would suggest

```suggestion
    public static function getCountryIsoCodeFromHeader(): ?string
```

And it would return the part matched in the regex

Reply:
will be done